docs/guides/dsm_azure_sobject.md

```diff
@@ -0,0 +1,288 @@
+    subscription_id = "0XXXXXXX-YYYY-HHHH-GGGG-123456789123"
+    key_vault_type = "STANDARD"
+  })
+}
+
+// Create an RSA security object in normal group
+resource "dsm_sobject" "rsa_key_dsm" {
+  name     = "rsa_key_dsm"
+  group_id = dsm_group.normal_group.id
+  key_size = 2048
+  key_ops = ["ENCRYPT", "DECRYPT", "WRAPKEY", "UNWRAPKEY", "SIGN", "VERIFY", "EXPORT"]
+  obj_type = "RSA"
+}
+
+// Copy above RSA security object to azure key vault
+resource "dsm_azure_sobject" "rsa_key_azure" {
+  name            = "rsa_key_azure"
+  group_id        = dsm_group.azure_group.id
+  key             = {
+    kid =  dsm_sobject.rsa_key_dsm.id
+  }
+  custom_metadata = {
+    azure-key-state =  "Enabled"
+    azure-key-name = "rsa-key-azure"
+  }
+  key_ops         = ["SIGN", "VERIFY", "ENCRYPT", "DECRYPT", "WRAPKEY", "UNWRAPKEY", "EXPORT", "APPMANAGEABLE", "HIGHVOLUME"]
+}
+
+// 1st Rotation of azure security object with AZURE option
+
+// Rotate RSA security object
+resource "dsm_sobject" "rsa_key_dsm_rotate1" {
+  name     = dsm_sobject.rsa_key_dsm.name
+  group_id = dsm_group.normal_group.id
+  key_size = 2048
+  key_ops  = ["ENCRYPT", "DECRYPT", "WRAPKEY", "UNWRAPKEY", "SIGN", "VERIFY", "EXPORT"]
+  obj_type = "RSA"
+  rotate   = "AZURE"
+  rotate_from = dsm_sobject.rsa_key_dsm.name
+}
+
+// Copy above RSA key to azure key vault
+resource "dsm_azure_sobject" "rsa_key_azure_rotate1" {
+  name            = dsm_azure_sobject.rsa_key_azure.name
+  group_id        = dsm_group.azure_group.id
+  key             = {
+    kid =  dsm_sobject.rsa_key_dsm_rotate1.id
+  }
+  custom_metadata = {
+    azure-key-state =  "Enabled"
+    azure-key-name = "rsa-key-azure"
+  }
+  key_ops         = ["SIGN", "VERIFY", "ENCRYPT", "DECRYPT", "WRAPKEY", "UNWRAPKEY", "EXPORT", "APPMANAGEABLE", "HIGHVOLUME"]
+  rotate   = "AZURE"
+  rotate_from = dsm_azure_sobject.rsa_key_azure.name
+}
+
+
+// 2nd Rotation of azure security object with AZURE option
+
+// Rotate RSA security object
+resource "dsm_sobject" "rsa_key_dsm_rotate2" {
+  name     = dsm_sobject.rsa_key_dsm.name
+  group_id = dsm_group.normal_group.id
+  key_size = 2048
+  key_ops  = ["ENCRYPT", "DECRYPT", "WRAPKEY", "UNWRAPKEY", "SIGN", "VERIFY", "EXPORT"]
+  obj_type = "RSA"
+  rotate   = "AZURE"
+  rotate_from = dsm_sobject.rsa_key_dsm.name
+}
+
+// Copy above RSA key to azure key vault
+resource "dsm_azure_sobject" "rsa_key_azure_rotate2" {
+  name            = dsm_azure_sobject.rsa_key_azure.name
+  group_id        = dsm_group.azure_group.id
+  key             = {
+    kid =  dsm_sobject.rsa_key_dsm_rotate2.id
+  }
+  custom_metadata = {
+    azure-key-state =  "Enabled"
+    azure-key-name = "rsa-key-azure"
+  }
+  key_ops         = ["SIGN", "VERIFY", "ENCRYPT", "DECRYPT", "WRAPKEY", "UNWRAPKEY", "EXPORT", "APPMANAGEABLE", "HIGHVOLUME"]
+  rotate   = "AZURE"
+  rotate_from = dsm_azure_sobject.rsa_key_azure.name
+}
+```
+
+## Soft deletion and Purge key of an Azure security object
```

Comment:
This section might have more impact if we provide the general `resource` definition and then below the code snippet have two sub points explaining the two additional fields that can be added to the definition. Not only does this remove the repeated portion, but it simplifies what the reader has to digest. 

If applicable to previous sections, I would strongly consider condensing some of it but if you feel those sections are adequate feel free to keep the examples as they are currently.

Reply:
Yes, I agree. So, in the examples of guide, I have given the least possible parameters.